kernel/src/main/java/org/kframework/parser/concrete2kore/disambiguation/PushTopAmbiguityUp.java

```diff
@@ -31,7 +31,7 @@ public Term apply(TermCons tc) {
     private class PushTopAmbiguityUp2 extends SafeTransformer {
         @Override
         public Term apply(TermCons tc) {
-            if (tc.production().klabel().isDefined() && tc.production().klabel().get().equals(KLabels.KREWRITE)) {
+            if (tc.production().klabel().isDefined() && tc.production().klabel().get().head().equals(KLabels.KREWRITE)) {
```

Comment:
Not sure why all these `.head()` are being inserted after `.klabel().get()` now. Why would there be multiple `klabel`? One per sort instantiation? How do we know the first is the correct one?

Reply:
This pr introduces a new method KLabel.head which returns the head of a klabel (ie its name without parameters). It's necessary because for example the klabel associated with the rewrite production has a parameter and thus otherwise will not compare equal to KLabels.KREWRITE.

Reply:
Ah, I see. Was thinking it was the head of a sequence.